Close btcpayserver#88 Added supporter logo

vaibhavgarg237 opened the pull request:
Added supporter logos
I created a JSON with { name, image, URL } and then mapped its content to render Supporters on the directory page...

Lemme know if you think there should be some edits...

vaibhavgarg237 reviewed (commented):
Rather than uploading images, we can also use URLs from the foundation directory. Lemme know what you think...

dennisreimann commented:
There is the [supporters sprite](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer/wwwroot/img/supporter-sprite.svg?short_path=eb1bdac), which works across light and dark mode. I think it'd be ok to copy its SVGG content, inline it within the `Supporters` component and then reference the logos like we do in the [app supporters partial](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer/Views/Shared/_BTCPaySupporters.cshtml).

Sidenote: It's a bit of a pain to maintain these across all the different repos, but there are subtle differences in how and where we display them, so I haven't found a good general approach for handling them.

dennisreimann commented:
Also: Let's see if we can arrrange it so that they visually have the same "weight" - e.g. Strike and HRF need to be wider for that. You can also take a look at the app supporters partial forr reference: We apply different widths to the logos, so that they look more balanced.

<img width="905" alt="Bildschirmfoto 2022-09-26 um 21 56 09" src="https://user-images.githubusercontent.com/886/192368880-00501390-3ce6-46ab-80f3-808127a786c6.png">

vaibhavgarg237 commented:
Removed [previous version](https://github.com/vaibhavgarg237/directory.btcpayserver.org/commit/9226b9bf10bcbe2514c3b65c04d4932041dce6ec) 
Previously I was using JSON, and a separate supporter.json file to map all the sponsors...
So that it works in dark mode too, added inline SVG's and adjusted their width so as they appear more balanced. I feel it's working as expected. 

Lemme know if you think it can be improved.

dennisreimann reviewed (changes requested):
Thanks! The logos work now, but I think they can be a bit smaller to fit the rest of the page. Also the responsive behaviour doesn't work quite right: The logos don't wrap nicely on mobile, which leads to the page having a horizontal scrollbar.

Can you adjust those things?

vaibhavgarg237 commented:
> 

Sry, I forgot to check for responsiveness. With current changes, I referred foundation repo, [github](https://github.com/btcpayserver/foundation.btcpayserver.org), [URL](https://foundation.btcpayserver.org/) for responsiveness and sizing. I feel it works fine now

Here is the hosted URL, to verify the changes...    https://addsponsor.netlify.app/

vaibhavgarg237 commented:
Sry, currently this still shows a horizontal scrollbar with a mobile view, I will adjust their width for the same...

dennisreimann reviewed (approved):
Yes, now it's working fine, thank you very much!
